chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -383,9 +383,22 @@ def unique_name
         end
 
         def claimed?
-          not @node[:crowbar_wall][:claimed_disks][self.unique_name][:owner].to_s.empty?
-        rescue
-          false
+          claimed_disks = @node[:crowbar_wall][:claimed_disks].to_hash
+          claimed_disks.each do | claimed_name, v |
+            claimed_real = File.realpath(claimed_name)
+            if claimed_real == self.name and not v[:owner].empty?
+              Chef::Log.info("Disk #{self.unique_name} (#{self.name}) already " \
+                             "claimed by #{claimed_name} (#{claimed_real}, " \
+                             "owner: #{v[:owner]})")
+              return true
+            end
+          end
+
+          begin
+            not @node[:crowbar_wall][:claimed_disks][self.unique_name][:owner].to_s.empty?
+          rescue
+            false
+          end
```

Comment:
> Well, you have that in your new code, don't you? So it shouldn't matter.

@vuntz Fixed in https://github.com/toabctl/barclamp-deployer/commit/ec18aba11adcf4231b02dfd3819ff39a1f8aa20c